From 20a3d2274134fa7336b3803882004e2f70a756de Mon Sep 17 00:00:00 2001
From: Leah <76408777+goldberl@users.noreply.github.com>
Date: Tue, 6 Jan 2026 15:01:59 -0500
Subject: [PATCH 2/4] fix: Pass interface string to get_newest_lease() (#6648)

RH-Author: Ani Sinha <anisinha@redhat.com>
RH-MergeRequest: 135: fix(cloudstack): Improve domain-name DHCP lease lookup
RH-Jira: RHEL-159033
RH-Acked-by: xiachen <xiachen@redhat.com>
RH-Acked-by: Miroslav Rezanina <mrezanin@redhat.com>
RH-Commit: [1/3] 5f74c14b51432facb96647410ed6a6ec52c99935 (anisinha/cloud-init)

In DataSourceCloudStack.py, get_newest_lease() is currently being
passed a Distro object, causing a Python type error. This PR
changes the Distro object to the interface string to fix this error.

(cherry picked from commit e02b739fc094eb8b56391705c2989f4aecbe139b)
Signed-off-by: Ani Sinha <anisinha@redhat.com>
---
 cloudinit/sources/DataSourceCloudStack.py | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/cloudinit/sources/DataSourceCloudStack.py b/cloudinit/sources/DataSourceCloudStack.py
index 61bf94f51..6d19077cd 100644
--- a/cloudinit/sources/DataSourceCloudStack.py
+++ b/cloudinit/sources/DataSourceCloudStack.py
@@ -314,7 +314,9 @@ def get_vr_address(distro):
             return latest_address
 
     with suppress(FileNotFoundError):
-        latest_lease = distro.dhcp_client.get_newest_lease(distro)
+        latest_lease = distro.dhcp_client.get_newest_lease(
+            distro.fallback_interface
+        )
         if latest_lease:
             LOG.debug(
                 "Found SERVER_ADDRESS '%s' via ephemeral %s lease ",
-- 
2.47.3

